types/datum.go

```diff
@@ -222,6 +222,9 @@ func (d *Datum) GetStringWithCheck(sc *stmtctx.StatementContext, chs string) (st
 
 func findEncoding(sc *stmtctx.StatementContext, chs string) (enc charset.Encoding, skip bool) {
 	enc = charset.FindEncoding(chs)
+	if sc == nil {
```

Comment:
When `sc` is nil?

Reply:
https://github.com/pingcap/tidb/pull/31196/files#diff-b7c54f7dad7257cbe4fa04b8d76c9bed694e2e2e7c5599f45eb2172d8d4ff8a7R913